Source/FieldSolver/SpectralSolver/SpectralFieldData.cpp

```diff
@@ -255,13 +255,17 @@ SpectralFieldData::BackwardTransform( MultiFab& mf,
                         mf_arr(i,j,k,i_comp) = inv_N*tmp_arr(i%nx, j%ny, k%nz);
                     });
             } else {
-                ParallelFor( mfi.validbox(),
-                [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-                    // Copy and normalize field
+                amrex::Box bx;
+                if (fill_guard_cells) {
+                    bx = mf[mfi].box();  // fill valid cells + guard cells
```

Comment:
Converted the PR to draft to do a bit more testing on the branch of #1051 in order to check whether I need to discard the last point in the nodal directions here, as it is done in the forward Fourier transform:
```suggestion
                    bx = mf[mfi].box();  // fill valid cells + guard cells
                    bx.enclosedCell();  // discard last point in nodal direction
```